### DISCUSSION TITLE
DAB install problems

### DISCUSSION THREAD

**Discussion by jshallBRI** [Azure/data-api-builder#1490] [Q&A] [answered]
I having difficulty installing/running DAB on my Windows machine.  

I originally attempted to use DAB through the SWA-CLI, but ran into errors (https://stackoverflow.com/questions/76244231/swa-cli-dataapibuilder-incompatible-with-windows-version).  In hopes of just working around the issue, I decided to try installing DAB independent of the SWA-CLI tool.

OS: Windows 10 Pro (10.0.19044 Build 19044)
VS Code: 1.78.2
PowerShell: 5.1.19041.2673
.Net: 6.0.408

When I try to install dab, I receive the following error:

`PS C:\Local Development\react> dotnet tool install --global Microsoft.DataApiBuilder
C:\Users\James98\AppData\Local\Temp\ad740bb1-0bf9-4d49-b292-dfd2d3a97ca4\restore.csproj : error NU1100: Unable to resolve 'microsoft.dataapibuilder (>= 0.0.0)' for 'net6.0'.
C:\Users\James98\AppData\Local\Temp\ad740bb1-0bf9-4d49-b292-dfd2d3a97ca4\restore.csproj : error NU1100: Unable to resolve 'microsoft.dataapibuilder (>= 0.0.0)' for 'net6.0/any'.
The tool package could not be restored.
Tool 'microsoft.dataapibuilder' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool`

I thought, "Maybe it is already installed?"  But an uninstall attempt yields the following:

`PS C:\Local Development\react> dotnet tool uninstall --global Microsoft.DataApibuilder
A tool with the package Id 'microsoft.dataapibuilder' could not be found. 

Tools are uninstalled using their package Id which may be different
from the tool name you use when calling the tool. You can find the tool names
and the corresponding package Ids for installed tools using the command
'dotnet tool list'.`

Any advice on how to fix this or what to try next would be GREATLY appreciated!

**Comment by Aniruddh25** [comment, marked as answer]
Hi @jshallBRI, Im sorry you are having trouble installing dab. 

Could you please try providing the nuget feed and version number explicitly: 

`dotnet tool install -g --add-source 'https://api.nuget.org/v3/index.json' Microsoft.DataApiBuilder --version 0.7.5`

**Reply by jshallBRI** [reply]
That worked perfectly!  Thanks, Aniruddh25!!

**Reply by Aniruddh25** [reply]
Im glad it worked for you! I'll close the discussion.
